app/components/CameraComponent.js

```diff
@@ -0,0 +1,132 @@
+import React, { Component } from 'react';
+import {
+  Alert,
+  Dimensions,
+  Image,
+  StyleSheet,
+  View,
+} from 'react-native';
+import Camera from 'react-native-camera';
+import Button from 'react-native-button';
+
+export default class CameraComponent extends Component {
+
+  state = {
+    image: null,
+  };
+
+  _takePicture() {
+    const setState = this.setState.bind(this);
```

Comment:
This shouldn't be necessary. Since you're using arrow functions, you should be able to just use `this.setState` in your Alert `onPress`